src/Contexts.ml

```diff
@@ -255,6 +259,10 @@ let ctx_lookup_type_decl (ctx : eval_ctx) (tid : TypeDeclId.id) : type_decl =
 let ctx_lookup_fun_decl (ctx : eval_ctx) (fid : FunDeclId.id) : fun_decl =
   FunDeclId.Map.find fid ctx.fun_context.fun_decls
 
+(** TODO: make this more efficient with maps *)
+let ctx_lookup_global_decl (ctx : eval_ctx) (gid : GlobalDeclId.id) : fun_decl =
+  ctx_lookup_fun_decl ctx (global_to_fun_id ctx.fun_context.gid_conv gid)
```

Comment:
Yes I was not convinced by my design choices (at the start because I wanted a first iteration that works with functions). I didn't touch much of it since last week with the graphs ^^'

Reply:
We now lookup the corresponding `global_decl`.

Reply:
Yes indeed, the TODO is obsolete